### PR TITLE
Update pdf-over from 4.2.2 to version :latest

### PR DESCRIPTION
**Pull request by suschizu**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [ ] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [ ] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Created with [`cask-repair`](https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md#updating-a-cask).

### CODE REVIEW

**Comment by core-code** [comment]
where does the URL come from? i don't see it on the homepage

File: Casks/pdf-over.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 cask "pdf-over" do
-  version "4.2.2"
```
Comment (vitorgalvao) [inline Casks/pdf-over.rb:2]: With no version in `url` and no `appcast`, this should be `version :latest` and `sha256 :no_checksum`.

But this is an unpopular app and cask which might get few updates, so keeping the version and checksum might not be problematic. So leave as is or make the change; whichever is more convenient to you. If you opt to leave the version and it becomes an issue later on, we can revisit it then.

**Comment by suschizu** [comment]
> where does the URL come from? i don't see it on the homepage

its the download from this site: https://webstart.buergerkarte.at/pdf-over/

**Comment by core-code** [comment]
> With no version in url and no appcast, this should be version :latest and sha256 :no_checksum.

i do now have a system which runs every 4 hours which is able to update casks with static downloads based on their contents. no appcast needed.

**Comment by core-code** [comment]
> > where does the URL come from? i don't see it on the homepage
> 
> its the download from this site: https://webstart.buergerkarte.at/pdf-over/

sorry i was blind, thought the version has been added not removed

**Comment by vitorgalvao** [comment]
> i do now have a system which runs every 4 hours which is able to update casks with static downloads based on their contents. no appcast needed.

That’s great! So it’d make sense to keep the version and checksum in this case, correct?

Does the system run on every cask? Would it make sense to switch to versioned more of the `version :latest` cases?

**Comment by core-code** [comment]
> That’s great! So it’d make sense to keep the version and checksum in this case, correct?

well, you can switch it over to 'version :latest' too if you prefer. but if the version number stays in there, my system should keep it up-to-date reliably.

> Does the system run on every cask? 

no it only runs on casks with a completely static (no #{} in the URL stanza) or effectively static ( just #{version.major}) URL. also i've put a few on a blacklist too, so its not 'every'. i've attached the current list.

> Would it make sense to switch to versioned more of the version :latest cases?

we can do that if we see a benefit there. for what i personally (actually professionally) want, "version :latest" cask are absolutely fine, except when the vendor also publishes beta versions which are not marked in any way. in this case a version number would be better and the new system should be able to ensure those are kept current easily.  

i do see issues with 'checksum mismatch' though in that 4-hour window. on the other hand, casks with appcast normally take longer to become updated too.

just for reference, the system does only re-download casks if the modification date or filesize of the download has changed.

[Uploading list.txt…]()

**Comment by core-code** [comment]
also note that the system is currently seriously hampered by the bugs in bump cask pr.

if 'cask-repair' could take a 'sha' flag to avoid re-downloading, i could use that one ^^

**Comment by vitorgalvao** [comment]
> no it only runs on casks with a completely static (no #{} in the URL stanza) or effectively static ( just #{version.major}) URL.

That’s what I meant to ask, if it ran on every applicable (unversioned) cask by default or some configuration was needed.

> we can do that if we see a benefit there.

There’s a major benefit for users, in that `upgrade` will work for the cask without `--greedy`. For us, knowing when and how often the software was updated (via cask version bumps) can also come in handy. The more casks we can have without `:latest`, the better.

> i do see issues with 'checksum mismatch' though in that 4-hour window. on the other hand, casks with appcast normally take longer to become updated too.

I think it will be fine for the overwhelmingly majority of casks. If it isn’t fine for a certain cask, we can revert that one.

> just for reference, the system does only re-download casks if the modification date or filesize of the download has changed.

Good to know. I was going to ask about the bandwidth usage.

> if 'cask-repair' could take a 'sha' flag to avoid re-downloading, i could use that one ^^

Added. It’s `-s`/`--cask-sha`. It’s untested, so do let me know if it’s not working.

**Comment by core-code** [comment]
> That’s what I meant to ask, if it ran on every applicable (unversioned) cask by default or some configuration was needed.

not really by default but we get a message for each cask that is not enabled for this and not on the blacklist either, so new additions should be picked up.

> There’s a major benefit for users, in that upgrade will work for the cask without --greedy
 
given the troubles HBC has with updating casks which have been highlighted by the recent 'zoom' rename i wonder if updating apps isn't something that isn't better handled by external updaters anyway :) 

not really being serious here of course

> I think it will be fine for the overwhelmingly majority of casks. If it isn’t fine for a certain cask, we can revert that one.

sure we can add more and see how it works out. i'll let you know when adding some to the blacklist because i cannot support it

> Added. It’s -s/--cask-sha. It’s untested, so do let me know if it’s not working.

awesome, will be sure to try it, thanks!

**Review by miccal — APPROVED** [review]

**Pull request merged**